### ISSUE TITLE
Bot is unable to detect stream live/offline status change

### ISSUE THREAD

**Issue by piebandit** [PhantomBot/PhantomBot#2527]
PhantomBot Version: 3.4.5
O/S Version: Linux: Ubuntu 18.04.4 LTS
Java Version: openjdk 1.8.0_292
Web Browser Version (if Panel related):

Detailed Description:
Bot does not seem to be able to detect stream live/offline status changes. Restarting the bot manually updates it to the current status, but won't update again until restarting the bot.

This has a snowball effect where anything that requires the channel be live won't function (automatic commercials, discord live notifications, etc.)

Steps to Reproduce:
N/A

Logs:
I did not see anything in the logs to indicate this error.

Screenshots (if applicable):

**Comment by argash** [comment]
I am having this issue as well. I have seen it in 3.4.2 and now 3.4.6

**Comment by Alucard2112** [comment]
Happened also to my version 3.4.6 Bot. Bot on 3.4.2 worked though.

**Comment by gmt2001** [comment]
Please ensure you have an `apioauth` token setup in your _botlogin.txt_, as the new API doesn't work without it. If you already have one, try re-doing it. If that still doesn't work, reply here again and we will move on with debugging.

**Comment by piebandit** [comment]
> Please ensure you have an `apioauth` token setup in your _botlogin.txt_, as the new API doesn't work without it. If you already have one, try re-doing it. If that still doesn't work, reply here again and we will move on with debugging.

I regenerated both tokens for kicks and it made no difference. I also turned on debug and helixdebug.
[debug-log.txt](https://github.com/PhantomBot/PhantomBot/files/6882307/debug-log.txt)

**Comment by argash** [comment]
FYI there is nothing in the setup guide about `apioauth `token

**Comment by argash** [comment]
My logs aren't complaining I don't think?

```
[07-27-2021 @ 03:52:48.670 GMT] Validated Twitch CHAT (oauth) OAUTH Token.
[07-27-2021 @ 03:52:48.670 GMT] Validated Twitch API (apioauth) OAUTH Token.
[07-27-2021 @ 03:52:49.334 GMT] Authenticated with Twitter API
```

**Comment by gmt2001** [comment]
> 
> 
> FYI there is nothing in the setup guide about `apioauth `token

It's listed as _Caster OAuth token_. I don't know who wrote it

**Comment by argash** [comment]
Just updated to 3.4.7 which did not fix my issue. However I noticed in another issue thread that you mentioned that someone should use the channel ID instead of the name. That actually worked (even though the channel name was different from the category name already).

**Comment by datadelft** [comment]
Not working here either on PhantomBot Version: 3.4.7 (stable) (Revision: 0b0f385)

**Comment by argash** [comment]
@gmt2001 well I thought that the 3.4.7 patch had fixed this issue, then last night after I went off line I noticed that it kept pushing "still streaming" tweets to twitter while I was asleep. However, on discord it only pushed the original go live notice.

**Comment by argash** [comment]
I just noticed that the bot is not updating it's status in discord. I finished my Friday stream around 2pm CDT yesterday and now at almost 11am CDT Saturday it still has the status that I'm streaming. Not sure if default logging will have any useful info for you but if it does let me know and I'll pull and post it for you.

**Comment by dungi** [comment]
I got this issue too, but the last 2 streams the detection seems to work again. Discord message got deleted and Bot is not "streaming". Maybe this was a Twitch Bug and it got fixed?

**Comment by gmt2001** [comment]
Okay, after much research I have implemented an undocumented PubSub topic that should provide near-real-time updates to Stream Up/Down status, as well as updating the viewer count at a similar rate, and with similar accuracy, to the Twitch website.

Notes:
- This is an ***undocumented*** PubSub topic, so there is a higher chance than normal of Twitch silently saying _fuck you_ to us
- No OAuth scopes are required to use this AFAIK, just a valid APIOAuth
- I have disabled the Stream Online/Offline notifications that came from TwitchAPI, as well as the viewer count from TwitchAPI, to prevent data conflicts, although they are still internally tracked
- Other variables that came from the TwitchAPI version of stream status tracking, such as uptime/start time, still rely on TwitchAPI
- This has ***no effect*** on `!promoteadm` in Discord
- This will effect the Online/Offline notifications from `!streamhandler` in Discord, but not the Game Change notifications
- Since some of the information comes from TwitchAPI, Online notifications in Discord may still show wrong Title/Game information, especially if Title/Game is not changed a few minutes before going online
- Since some of the information comes from TwitchAPI, Offline notifications in Discord may still show wrong chatters/followers counts

This will deploy to the next nightly in a few hours. Exceptions will be automatically sent via Rollbar, but let me know if there are any non-exception issues.
